pandas/core/arrays/period.py

```diff
@@ -92,22 +94,44 @@ def wrapper(self, other):
             self._check_compatible_with(other)
 
             result = ordinal_op(other.ordinal)
-        elif isinstance(other, cls):
-            self._check_compatible_with(other)
-
-            result = ordinal_op(other.asi8)
-
-            mask = self._isnan | other._isnan
-            if mask.any():
-                result[mask] = nat_result
 
-            return result
         elif other is NaT:
             result = np.empty(len(self.asi8), dtype=bool)
             result.fill(nat_result)
-        else:
+
+        elif not is_list_like(other):
```

Comment:
can you consolidate this with the is_like_check at the very begining.

any way to simplify this logic (or is that waiting for combinng later)?

Reply:
e.g. on line 81 you have a scalar, else you have a list-like

Reply:
> or is that waiting for combinng later

Yah, at the moment we have both cosmetic and non-cosmetic differences between the DTA/TDA/PA methods.  I'm addressing the non-cosmetic ones first.